sdk/v2/authx/roles.go

```diff
@@ -59,3 +60,71 @@ func (r RoleAssignment) MarshalJSON() ([]byte, error) {
 		},
 	)
 }
+
+// RoleAssignmentsClient is the specialized client for managing RoleAssignments
+// with the Brigade API.
+type RoleAssignmentsClient interface {
+	// Grant grants the system-level Role specified by the RoleAssignment to the
+	// principal also specified by the RoleAssignment.
+	Grant(context.Context, RoleAssignment) error
+	// Revoke revokes the system-level Role specified by the RoleAssignment for
+	// principal also specified by the RoleAssignment.
+	Revoke(context.Context, RoleAssignment) error
+}
+
+type roleAssignmentsClient struct {
+	*rm.BaseClient
+}
```

Comment:
This is the client for system-level role assignments. Project-level role assignments are special, so they have their own separate client that hangs off the projects client.